crates/aptos/src/common/types.rs

```diff
@@ -1090,6 +1125,14 @@ pub struct MovePackageDir {
     /// Select bytecode, language, compiler for Move 2
     #[clap(long)]
     pub move_2: bool,
+
+    /// Select optimization level
+    #[clap(long, value_parser = clap::value_parser!(OptimizationLevel))]
+    pub optimization_level: Option<OptimizationLevel>,
+
+    /// Enable lint
+    #[clap(long)]
+    pub lint: bool,
```

Comment:
So can I lint when I do other actions like publish packages?  Is this the right place to put this or have a separate command

Reply:
Currently, lint is carried out by certain passes in the compiler, so we must run the compiler to do it.  As with many existing lint programs, we can provide warnings/errors (if appropriate) as well as the lint warnings.  Unlike many, our current codebase seems to require us to generate code output (but not metadata) to run all checks (extended_checks), so perhaps we should leave this as a compiler option.  Something to discuss, now that the functionality is there.